From a04411f8a00bb111612e48859140612ce2ed0289 Mon Sep 17 00:00:00 2001
From: Julien Cristau <jcristau@debian.org>
Date: Mon, 15 Nov 2010 17:09:07 +0100
Subject: [PATCH] Revert "display: outputs are enabled automatically by KMS"

This reverts commit 19c48d3b3f33582baa87a9b3a9189e320e4cea45.

Turns out they aren't, and this resets the backlight to 0 on a mode
switch.  Debian bug#600405.
---
 src/intel_display.c |   16 ++++++++++++++--
 1 files changed, 14 insertions(+), 2 deletions(-)

diff --git a/src/intel_display.c b/src/intel_display.c
index d32224e..d9c4b5b 100644
--- a/src/intel_display.c
+++ b/src/intel_display.c
@@ -327,7 +327,7 @@ intel_crtc_apply(xf86CrtcPtr crtc)
 	uint32_t *output_ids;
 	int output_count = 0;
 	int fb_id, x, y;
-	int i, ret = FALSE;
+	int i, ret;
 
 	output_ids = calloc(sizeof(uint32_t), xf86_config->num_output);
 	if (!output_ids)
@@ -378,14 +378,26 @@ intel_crtc_apply(xf86CrtcPtr crtc)
 	} else
 		ret = TRUE;
 
+	/* Turn on any outputs on this crtc that may have been disabled */
+	for (i = 0; i < xf86_config->num_output; i++) {
+		xf86OutputPtr output = xf86_config->output[i];
+
+		if (output->crtc != crtc)
+			continue;
+
+		intel_output_dpms(output, DPMSModeOn);
+	}
+
 	intel_set_gem_max_sizes(scrn);
 
 	if (scrn->pScreen)
 		xf86_reload_cursors(scrn->pScreen);
 
+	return ret;
+
 done:
 	free(output_ids);
-	return ret;
+	return FALSE;
 }
 
 static Bool
-- 
1.7.2.3

